### PR TITLE
Adds Project Member API endpoint

### PR DESCRIPTION
**Pull request by beerlington**
This will be used to list all users on a project, add new members, and
revoke membership. The ProjectMember model acts as both a linking table
between users and projects, but also the invitation model.

This does not yet include endpoints related to accepting or resending
invitations.

UI - https://github.com/AgilionApps/LaunchPad-UI/pull/14

### CODE REVIEW

File: app/controllers/v1/oauth_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def status
   end
 
   def create
-    user = User.find_by("lower(email) = ?", params[:username].downcase)
+    user = User.find_by_email(params[:username])
```
Comment (patrickberkeley) [inline app/controllers/v1/oauth_controller.rb:9]: :+1:

File: app/controllers/v1/project_members_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  def create
+    project = current_user.projects.find(params[:project_member][:project_id])
+    member = project.members.new(create_params)
+
+    if member.save_and_send_invite
+      render json: member, serializer: ProjectMemberSerializer, status: :created
+    else
+      render json: {errors: member.errors}, status: :unprocessable_entity
```
Comment (patrickberkeley) [inline app/controllers/v1/project_members_controller.rb]: There's helper method for errors (via Alan) https://github.com/AgilionApps/LaunchPad-API/blob/master/app/controllers/application_controller.rb#L23-L25

File: app/models/project.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class Project < ActiveRecord::Base
-  has_many :project_members
-  has_many :users, through: :project_members
+  has_many :members, class_name: 'ProjectMember'
```
Comment (patrickberkeley) [inline app/models/project.rb:2]: :+1:

**Comment by patrickberkeley** [comment]
LGTM

File: app/controllers/v1/project_members_controller.rb (hunk 2 above)
Comment (beerlington) [inline app/controllers/v1/project_members_controller.rb]: awesome, thanks

**Pull request merged**